DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,179 @@
+    ```
+
+## Usage with preexisting databases
+
+If you want to use Mathesar with a preexisting Postgres DB, modify the `DATABASES.mathesar_tables` entry of the `config/settings.py` file with appropriate connection details before installing the Mathesar types and functions by running `install.py` as described in the previous step.
+
+## Rebuilding the Docker images
+
+Sometimes you may need to rebuild your Docker images after pulling new code changes or switching branches. Do so via:
+
+```
+docker compose --profile dev up --force-recreate --build dev-service
+```
+
+
+## Demo mode
+
+Mathesar can be run in "demo mode" to meet the specific needs of our [live demo site](https://demo.mathesar.org).
+
+See our [Live demo mode](./demo/README.md) guide for more information on enabling live demo mode locally
+
+
+## Opening a shell in the container
+
+- If you need to do some work within the container you can open a bash shell via:
+
+    ```
+    docker exec -it mathesar_service_dev bash
+    ```
+
+- To open a PostgreSQL [psql](https://www.postgresql.org/docs/current/app-psql.html) terminal for the data in Mathesar:
+
+    ```
+    docker exec -it mathesar_db psql -U mathesar
+    ```
+
+## Troubleshooting
+
+### Permissions within Windows
+
+- Running Script in powershell is disabled by default in windows , you have to change permission to run scripts  [Official Docs ](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.security/set-executionpolicy?view=powershell-7.2) 
+
+### Fixing line endings on Windows
+
+If you happen to clone the git repository outside of WSL, then you fix it by running these commands from within WSL.
+
+```
+git config --global core.autocrlf input
+sudo apt-get install -y dos2unix
+sudo find . -type f -exec dos2unix {} \;
+```
+
+These commands install the `dos2unix` utility, which converts text files from the DOS/Microsoft Windows format (with CRLF line endings) to the Unix/Linux format (with LF line endings). Next, the find utility is used to locate all files (-type f) in the current directory (.) and its subdirectories, and the dos2unix command is then executed on each of them (-exec dos2unix ;).
+
+
+### Live reloading front end code on Windows
+
+Hot module replacement for front end code does not work when the project is present on a windows filesystem and WSL is used to run docker. [Read more](./mathesar_ui/README.md#live-reloading-on-windows).
+
+### PostgreSQL server is running on your host machine
+
+* If you you see the following error after running `docker-compose up`, then the port used by Postgres is already in use.
+
+    > ERROR: for db Cannot start server db: driver failed programming external connectivity on endpoint mathesar_db (70c521f468cf2bd54014f089f0051ba28e2514667): Error starting userland proxy: listen tcp4 0.0.0.0:5432: bind: address already in use.
+
+* Try stopping Postgres with:
+
+  ```
+  sudo service postgresql stop
+  ```
+
+ - Then run `docker-compose up` again.
```

Comment:
Same here; this is imprecise.

Reply:
Also addressed in 9fadb6011a1ac800a620a83a761bacd468bc2d8a.